src/azure-cli-testsdk/azure/cli/testsdk/base.py

```diff
@@ -137,6 +139,9 @@ def _merge_lists(base, patches):
     def tearDown(self):
         for processor in self._processors_to_reset:
             processor.reset()
+        if self.random_config_dir:
+            from azure.cli.core.util import rmtree_with_retry
+            rmtree_with_retry(self.cli_ctx.config.config_dir)
```

Comment:
`TemporaryDirectory` has `cleanup()` method, but I don't want to save this instance into `cli_ctx`, use rmtree instead.